pyuvdata/uvdata/uvdata.py

```diff
@@ -2360,6 +2380,167 @@ def _check_freq_spacing(self, raise_errors=True):
+            Default is False.
+        raise_warning : bool
+            If an object cannot be converted to flex-pol, and `raise_error=False`, then
+            raise a warning that the conversion failed. Default is True.
+        """
+        if not self.flex_spw:
+            msg = "Cannot make a flex-pol UVData object if flex_spw=False."
+            if raise_error:
+                raise ValueError(msg)
+            if raise_warning:
+                warnings.warn(msg)
+            return
+
+        if self.Npols == 1:
+            # This is basically a no-op, fix the to pol-array attributes and exit
+            if self.flex_spw_polarization_array is None:
+                self.flex_spw_polarization_array = (
+                    np.zeros_like(self.spw_array) + self.polarization_array[0]
+                )
+                self.polarization_array = np.array([0])
+            return
+
+        flex_pol_idx = np.zeros_like(self.spw_array)
+        for idx, spw in enumerate(self.spw_array):
+            spw_screen = self.flex_spw_id_array == spw
+
+            if self.future_array_shapes:
+                pol_check = ~np.all(self.flag_array[:, spw_screen], axis=(0, 1))
+            else:
+                pol_check = ~np.all(self.flag_array[:, 0, spw_screen], axis=(0, 1))
+
+            if sum(pol_check) > 1:
+                msg = (
+                    "Cannot make a flex-pol UVData object, as some windows have "
+                    "unflagged data in mutiple polarizations."
+                )
+                if raise_error:
+                    raise ValueError(msg)
+                if raise_warning:
+                    warnings.warn(msg)
+                return
+            elif not np.any(pol_check):
+                flex_pol_idx[idx] = -1
+            else:
+                flex_pol_idx[idx] = np.where(pol_check)[0][0]
+
+        # If one window was all flagged out, but the others all belong to the same pol,
+        # assume we just want that polarization.
+        if len(np.unique(flex_pol_idx[flex_pol_idx >= 0])) == 1:
+            flex_pol_idx[:] = np.unique(flex_pol_idx[flex_pol_idx >= 0])
+
+        # Now that we have polarizations sorted out, update metadata attributes
+        self.flex_spw_polarization_array = self.polarization_array[flex_pol_idx]
+        self.polarization_array = np.array([0])
+        self.Npols = 1
+
+        # Finally, prep for working w/ data-like attibutes. Start by determining the
+        # right shape for the new values, modulo future_array_shapes
+        new_shape = [self.Nblts, 1, self.Nfreqs, 1]
+
+        # Pop the spw-axis if using future array shapes
+        if self.future_array_shapes:
+            del new_shape[1]
+
+        # Now go through one-by-one with data-like parameters and update
+        for name, param in zip(self._data_params, self.data_like_parameters):
+            # Use fill here, since we want zeros if we are working with nsample_array
```

Comment:
again says "fill" but uses empty()

Reply:
Ah, good catch -- old code used "fill", until I realized I was writing to every single value in the array, hence why I switched to empty.